src_c/math.c

```diff
@@ -2082,6 +2084,40 @@ static PyBufferProcs vector_as_buffer = {
 };
 #endif
 
+static PyObject *
+vector___round__(pgVector *self, PyObject *args)
+{
+    Py_ssize_t i, ndigits;
+    PyObject *o_ndigits = NULL;
+
+    pgVector *ret = _vector_subtype_new(self);
+
+    if (ret == NULL || !PyArg_ParseTuple(args, "|O", &o_ndigits)) {
+        return NULL;
+    }
+
+    memcpy(ret->coords, self->coords, sizeof(ret->coords[0]) * ret->dim);
+
+    if (o_ndigits == NULL || o_ndigits == Py_None) {
+        for (i = 0; i < ret->dim; ++i)
+            ret->coords[i] = round(ret->coords[i]);
+    }
+    else if (RealNumber_Check(o_ndigits)) {
+        ndigits = PyNumber_AsSsize_t(o_ndigits, NULL);
+        if (ndigits == -1 && PyErr_Occurred())
```

Comment:
Does it need that `ndigits == -1` part?

Reply:
I'm not sure why you're asking this. That's what the function says it will do on error, seems reasonable to check that exactly-- https://docs.python.org/3/c-api/number.html#c.PyNumber_AsSsize_t.